Packages/com.unity.inputsystem/InputSystem/Editor/Analytics/InputActionCodeAuthoringAnalytic.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            Debug.Log("Play mode state change: " + change);
+            if (change == PlayModeStateChange.ExitingEditMode)
+            {
+                // Reset all counters when exiting edit mode
+                for (var i = 0; i < m_Counters.Length; ++i)
+                    m_Counters[i] = 0;
+
+                // Make sure not suppressed
+                suppress = false;
+            }
+            if (change == PlayModeStateChange.ExitingPlayMode)
+            {
+                // Send analytics and unhook delegate when exiting play-mode
+                EditorApplication.playModeStateChanged -= PlayModeChanged;
+                new InputActionCodeAuthoringAnalytic().Send();
+
+                // No reason to not suppress
+                suppress = true;
+            }
+        }
+
+        [InitializeOnEnterPlayMode]
+        private static void Hook()
+        {
+            // Make sure only a single play-mode change delegate is registered
+            EditorApplication.playModeStateChanged -= PlayModeChanged;
+            EditorApplication.playModeStateChanged += PlayModeChanged;
+        }
+
+        private InputActionCodeAuthoringAnalytic()
+        {
+            info = new InputAnalytics.InputAnalyticInfo(kEventName, kMaxEventsPerHour, kMaxNumberOfElements);
+        }
+
+        /// <summary>
+        /// Represents InputAction code authoring editor data.
+        /// </summary>
+        /// <remarks>
+        /// Ideally this struct should be readonly but then Unity cannot serialize/deserialize it.
+        /// </remarks>
+        [Serializable]
+        public struct Data : UnityEngine.InputSystem.InputAnalytics.IInputAnalyticData
+        {
+            /// <summary>
+            /// Creates a new <c>Data</c> instance.
+            /// </summary>
+            /// <param name="usesCodeAuthoring">Specifies whether code authoring has been used.</param>
+            public Data(bool usesCodeAuthoring)
+            {
+                uses_code_authoring = usesCodeAuthoring;
+            }
+
+            /// <summary>
+            /// Specifies whether code-authoring (Input Action setup via extensions) was used at least once
+            /// during play-mode.
+            /// </summary>
+            public bool uses_code_authoring;
+        }
+
+#if UNITY_2023_2_OR_NEWER
+        public bool TryGatherData(out UnityEngine.Analytics.IAnalytic.IData data, out Exception error)
+#else
+        public bool TryGatherData(out InputAnalytics.IInputAnalyticData data, out Exception error)
+#endif
+        {
+            try
+            {
+                // Determine aggregated perspective, i.e. was "any" code-authoring API used
+                var usedCodeAuthoringDuringPlayMode = false;
+                for (var i = 0; i < m_Counters.Length; ++i)
+                {
+                    if (m_Counters[i] > 0)
+                    {
+                        usedCodeAuthoringDuringPlayMode = true;
```

Comment:
setting this bool in Register() would save the cost of going through the array here.

Reply:
Yes that is a valid point. However, I created the counters to allow tracking in more detail at a later point. Hence I intend to leave this as-is for now. If I would change it, it would suggest removing the counters, which would suggest all registrations should not be specific etc. This should not have much cost to bother about?

Reply:
(One reason for that decision was that it was a very manual process to go through all method calls and stacked method calls to established the detailed information about what is being used).

Reply:
Okay, that makes sense